git/readtree.go

```diff
@@ -495,18 +502,63 @@ func readtreeSaveIndex(c *Client, opt ReadTreeOptions, i *Index) error {
 	return nil
 }
 
+// Check if the read-tree can be performed. Returns a list of files that need to be
+// removed if --reset is specified.
+func checkReadtreePrereqs(c *Client, opt ReadTreeOptions, idx *Index) ([]File, error) {
+	if opt.Update && opt.Prefix == "" && !opt.Merge && !opt.Reset {
+		return nil, fmt.Errorf("-u is meaningless without -m, --reset or --prefix")
```

Comment:
It seems a little weird to produce error messages in the git package that refer to flags defined in the cmd package. I'm assuming that this is here because of the official git tests checking specific error messages?

Reply:
I can't remember if there was a test requiring the wording or I just wanted it to match,  but this is just moving it from the end in checkMergeAndUpdate to a new checkPrereqs to evaluate the cheap checks before doing all the work.